internal/cmd/operator-sdk/generate/bundle/bundle.go

```diff
@@ -297,49 +298,103 @@ func (c bundleCmd) runMetadata() error {
 }
 
 // findRelatedImages looks in the controller manager's environment for images used by the operator.
-func (c bundleCmd) findRelatedImages(col *collector.Manifests) (map[string]string, error) {
-	const relatedImagePrefix = "RELATED_IMAGE_"
-	env, err := c.findManagerEnvironment(col)
+func (c bundleCmd) findRelatedImages(col *collector.Manifests) ([]operatorsv1alpha1.RelatedImage, error) {
+	relatedImages, relatedImageSources, err := c.collectRelatedImages(col)
 	if err != nil {
 		return nil, err
 	}
-	imageNames := make(map[string]string, len(env))
-	for _, envVar := range env {
-		if strings.HasPrefix(envVar.Name, relatedImagePrefix) {
-			if envVar.ValueFrom != nil {
-				return nil, fmt.Errorf("related images with valueFrom field unsupported, found in %s`", envVar.Name)
+
+	var (
+		finalRelatedImages = make([]operatorsv1alpha1.RelatedImage, 0, len(relatedImages))
+		skip               = make(map[int]struct{})
+	)
+
```

Comment:
I wonder if there's another way to organize this to avoid O(n^2).

Would this organization help:
```
type relatedImage struct {
    containerID string
    name string
}

relatedImagesByImageRef := map[string][]relatedImage{}
```

Then for each key in the map (the image reference), we know we'll generate exactly one `operatorsv1alpha1.RelatedImage`, and to generate that one, all we need to look at are the values of that key (not all the values of all the keys)